presto-kafka/src/main/java/io/prestosql/plugin/kafka/encoder/json/format/ISO8601DateTimeFormatter.java

```diff
@@ -0,0 +1,90 @@
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.prestosql.plugin.kafka.encoder.json.format;
+
+import io.prestosql.spi.type.SqlDate;
+import io.prestosql.spi.type.SqlTime;
+import io.prestosql.spi.type.SqlTimeWithTimeZone;
+import io.prestosql.spi.type.SqlTimestamp;
+import io.prestosql.spi.type.SqlTimestampWithTimeZone;
+
+import java.time.Instant;
+import java.time.LocalDate;
+import java.time.LocalDateTime;
+import java.time.LocalTime;
+import java.time.ZonedDateTime;
+
+import static io.prestosql.plugin.base.type.DateTimes.NANOSECONDS_PER_MICROSECOND;
+import static io.prestosql.plugin.base.type.DateTimes.NANOSECONDS_PER_MILLISECOND;
+import static io.prestosql.plugin.base.type.DateTimes.PICOSECONDS_PER_NANOSECOND;
+import static io.prestosql.plugin.base.type.DateTimes.getMicrosOfSecond;
+import static io.prestosql.plugin.base.type.DateTimes.getMillisOfSecond;
+import static io.prestosql.plugin.base.type.DateTimes.roundToNearest;
+import static io.prestosql.plugin.base.type.DateTimes.scaleEpochMicrosToSeconds;
+import static io.prestosql.plugin.base.type.DateTimes.scaleEpochMillisToSeconds;
+import static io.prestosql.plugin.kafka.encoder.json.format.util.TimeConversions.getNanosOfDay;
+import static io.prestosql.plugin.kafka.encoder.json.format.util.TimeConversions.localDateTimeOfEpochNanos;
+import static io.prestosql.plugin.kafka.encoder.json.format.util.TimeConversions.scaleEpochMillisToNanos;
+import static io.prestosql.plugin.kafka.encoder.json.format.util.TimeConversions.scaleEpochPicosToNanos;
+import static java.time.ZoneOffset.UTC;
+import static java.time.format.DateTimeFormatter.ISO_OFFSET_DATE_TIME;
+import static java.time.format.DateTimeFormatter.ISO_OFFSET_TIME;
+
+public class ISO8601DateTimeFormatter
+        extends UnimplementedJsonDateTimeFormatter
+{
+    @Override
+    public String formatDate(SqlDate value)
+    {
+        return LocalDate.ofEpochDay(value.getDays()).toString();
+    }
+
+    @Override
+    public String formatTime(SqlTime value, int precision)
+    {
+        return LocalTime.ofNanoOfDay(getNanosOfDay(scaleEpochPicosToNanos(value.getPicos()))).toString();
+    }
+
+    @Override
+    public String formatTimeWithZone(SqlTimeWithTimeZone value)
+    {
+        return ISO_OFFSET_TIME.format(
+                ZonedDateTime.of(localDateTimeOfEpochNanos(scaleEpochMillisToNanos(value.getMillisUtc())), UTC));
+    }
+
+    @Override
+    public String formatTimestamp(SqlTimestamp value)
+    {
+        long epochMicros = value.getEpochMicros();
+        long picosOfMicros = value.getPicosOfMicros();
+
+        long epochSecond = scaleEpochMicrosToSeconds(epochMicros);
+        int nanoFraction = getMicrosOfSecond(epochMicros) * NANOSECONDS_PER_MICROSECOND + (int) (roundToNearest(picosOfMicros, PICOSECONDS_PER_NANOSECOND) / PICOSECONDS_PER_NANOSECOND);
```

Comment:
`roundToNearest(picosOfMicros, PICOSECONDS_PER_NANOSECOND) / PICOSECONDS_PER_NANOSECOND` - this is the same as `roundDiv(picosOfMicros, PICOSECONDS_PER_NANOSECOND)`, right?  Perhaps you can change its visibility.  I found myself needing it too.

Reply:
I think this is the same as scaling from picoseconds to nanoseconds precision